### PR TITLE
React native picker

### PR DESCRIPTION
**Pull request by abdallahshaban557**
_Issue #, if available:_

_Description of changes:_

Changes to add React Native as a platform picker to the libraries section. Took this opportunity to clean up the documentation as I progressed. This is still a draft.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/utils/filter-data.ts
Comment (tjleing) [inline src/utils/filter-data.ts:25]: `react-native` is in `FRAMEWORK_FILTER_OPTIONS` so this results in duplications in `INTEGRATION_FILTER_OPTIONS`.  Generally all keywords up to now have fit neatly into either (web) frameworks (`/ui`?) or platforms (`/lib`, `/guides`) but it seems like the goal of this change is to make React Native a member of both.  This can have a lot of effects, for example I would expect RN to appear twice on the getting-started pages' dropdown.  Can we have a preview docs site for this @timngyn?

**Comment by tjleing** [comment]
also looks like the build is failing on GH with this error:
```
./src/fragments/lib/datastore/native_common/setup-env.mdx
Module not found: Can't resolve '/src/fragments/lib/datastore/android/getting-started/30_platformIntegration.mdx' in 
```

File: src/directory/directory.js
Comment (tjleing) [inline src/directory/directory.js:489]: it looks like in-app messaging is available for JS and RN but the UI components are only available for RN, is this correct?  do we want to remove JS support here (or did you move it somewhere else)?

File: src/utils/filter-data.ts
Comment (tjleing) [reply src/utils/filter-data.ts:25]: I guess it wouldn't affect the dropdown since that just reads from the directory list, but we should still look to make sure that this change doesn't have any accidental effects.

Comment (abdallahshaban557) [reply src/utils/filter-data.ts:25]: I tried to double check to make sure it doesn't cause any other issues. But will leave it up to @timngyn to see if this should be done differently.

File: src/directory/directory.js
Comment (abdallahshaban557) [reply src/directory/directory.js:489]: Without the UI components the category is exceptionally hard to use. For now I'm gonna hide it until we go GA end of september and show it for JS (specifically React).

File: src/utils/filter-data.ts
Comment (tjleing) [reply src/utils/filter-data.ts:25]: there's a few edge cases that are slowly coming back to me now that I'm looking at the codebase again 😅  for example, ChooseFilterPages like [/start/](https://docs.amplify.aws/start/) (with cookies cleared/in incognito) that I would want to just check briefly if we had a preview site up.  if we do find an issue there just something like `[...new Set([...PLATFORM_FILTER_OPTIONS, ...FRAMEWORK_FILTER_OPTIONS])]` instead of the currently existing
```
const INTEGRATION_FILTER_OPTIONS = [
  ...PLATFORM_FILTER_OPTIONS,
  ...FRAMEWORK_FILTER_OPTIONS,
];
```
would certainly work.  hopefully I'm just being paranoid over nothing here 😄

**Comment by tjleing** [comment]
if you select RN as an integration on `/start` it still sets your filterKeys platform to `js` (image below) so if you then pick `/lib` it redirects to JS instead of RN — the logic is in [getFilterOverrides.ts](https://github.com/aws-amplify/docs/blob/c59fb954e5edbd7ca7a765d4bebbd8f2395fba60/src/utils/withFilterOverrides.ts).  looks to me like adding `&& updates.integration !== "react-native"` to line 24 should do it but not totally sure

image

**Review by cshfang — COMMENTED** [review]
Looking good. Had some comments and suggestions. Thanks so much for doing all this.

File: src/pages/lib/analytics/identifyuser/q/platform/[platform].mdx
Comment (abdallahshaban557) [reply src/pages/lib/analytics/identifyuser/q/platform/[platform].mdx:6]: got it! I have a bucket of enhancements for analytics that I will add this item to. Thanks @cshfang !

File: src/fragments/lib/analytics/ios/existing-resources.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/analytics/ios/existing-resources.mdx]: Fixed

File: src/fragments/lib/analytics/android/existing-resources.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/analytics/android/existing-resources.mdx:27]: Fixed

File: src/fragments/lib/reactnative.mdx
Comment (timngyn) [inline src/fragments/lib/reactnative.mdx:1]: Should this file be updated to reflect React Native? I think at the moment this is similar to what we have for Javascript

File: src/fragments/lib/reactnative.mdx
Comment (timngyn) [inline src/fragments/lib/reactnative.mdx:16]: When I click `Authentication`, `Geo`, and `AR/VR` it says that React Native isn't supported. Should these links be removed or do we also need to update the docs to support React Native?

**Review by timngyn — COMMENTED** [review]
Awesome job @abdallahshaban557!!

I didn't go through all the content changes, did have some small comments.

I created a PR here to merge into your branch with the suggestions from T.J. and an update to include the React Native  "Libraries" index page: https://github.com/aws-amplify/docs/pull/4651

File: src/fragments/lib/reactnative.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/reactnative.mdx:1]: Kept this the same since technically it is still using the Javascript library.

File: src/fragments/lib/reactnative.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/reactnative.mdx:16]: Geo and AR/VR should not be shown. I am not sure what happened with Authentication - that was all working!!! checking!

Comment (abdallahshaban557) [reply src/fragments/lib/reactnative.mdx:16]: Fixed Auth right now - thanks for finding that. I think I had overwritten some of my changes. Auth is my next enormous challenge after this PR!

File: src/fragments/lib/datastore/android/getting-started/30_platformIntegration.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/datastore/android/getting-started/30_platformIntegration.mdx:1]: yes, I kept it hidden and removed it for now. It was jarring to just have something that said coming soon.

File: src/fragments/lib/analytics/js/update-endpoint.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/analytics/js/update-endpoint.mdx]: Fixed

File: src/fragments/lib/analytics/js/update-endpoint.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/analytics/js/update-endpoint.mdx:58]: Will fix at a later time. I want to focus on Auth first.

File: src/fragments/lib/ios.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/ios.mdx:3]: Yes, I am adding those for myself as breadcrumbs to go in at a later time to fix with the docs team.

File: src/fragments/lib/pubsub/js/getting-started.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/pubsub/js/getting-started.mdx:10]: Fixed!

File: src/fragments/lib/restapi/js/delete.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/restapi/js/delete.mdx:9]: Fixed!

File: src/fragments/lib/restapi/js/fetch.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/restapi/js/fetch.mdx:9]: Fixed

File: src/fragments/lib/restapi/js/update.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/restapi/js/update.mdx]: Fixed

File: src/fragments/lib/analytics/js/existing-resources.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/analytics/js/existing-resources.mdx:96]: I am not sure - I am adding a TODO to check that at a later time.

**Review by timngyn — APPROVED** [review]

File: src/fragments/lib/analytics/js/update-endpoint.mdx
Comment (cshfang) [reply src/fragments/lib/analytics/js/update-endpoint.mdx:60]: I just meant there should be a line break after the closing tag but it's nbd

**Review by cshfang — APPROVED** [review]

**Review by tjleing — DISMISSED** [review]

**Review by chintannp — COMMENTED** [review]
We need a separate `src/fragments/lib/storage/js/upload.mdx` file for React Native which is a bit different than JS. The **browser upload section** needs to be removed for React Native. Additionally, I think the "React Native uploads" section needs to be renamed to "Media Uploads" or something similar to make more sense.

**Review by rachelnabors — CHANGES_REQUESTED** [review]
I would like to see @chintannp's content requests met before approving this, thanks!

**Comment by abdallahshaban557** [comment]
Both of the issues mentioned by @chintannp have been addressed. Can I please get approvals on this PR now? @chintannp @rachelnabors

File: src/fragments/lib/storage/js/react-native-uploads.mdx
Comment (chintannp) [inline src/fragments/lib/storage/js/react-native-uploads.mdx:1]:
```suggestion
## Media uploads

Upload an image in your React Native app:
```

File: src/fragments/lib/storage/js/upload.mdx
Comment (chintannp) [inline src/fragments/lib/storage/js/upload.mdx:11]:
```suggestion
```

File: src/fragments/lib/storage/js/browser-uploads.mdx
Comment (chintannp) [inline src/fragments/lib/storage/js/browser-uploads.mdx:1]:
```suggestion
## Browser uploads

Upload an image in the browser:
```

Comment (chintannp) [reply src/fragments/lib/storage/js/browser-uploads.mdx:1]: For JS, we need to keep title to `Browser uploads` because the code example is for browser uploads. For RN only, we need to update it to `Media Uploads`.

File: src/fragments/lib/storage/js/react-native-uploads.mdx
Comment (abdallahshaban557) [reply src/fragments/lib/storage/js/react-native-uploads.mdx:1]: I merged both titles and have that available in the main uploads file now - please check my previous commit. I am closing out this suggestion.

**Comment by abdallahshaban557** [comment]
@chintannp - I have resolved your new comments! Please let me know if there is anything else for me to fix!

**Review by rachelnabors — APPROVED** [review]

**Review by chintannp — APPROVED** [review]
LGTM 🚀

**Pull request merged**
